pkg/controller/daemon/daemon_controller.go

```diff
@@ -774,9 +774,14 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ds *apps.DaemonSet) (map[s
 	// Group Pods by Node name.
 	nodeToDaemonPods := make(map[string][]*v1.Pod)
 	for _, pod := range claimedPods {
-		nodeName := pod.Spec.NodeName
+		nodeName, err := util.GetTargetNodeName(pod)
+		if err != nil {
+			return nil, err
+		}
+
 		nodeToDaemonPods[nodeName] = append(nodeToDaemonPods[nodeName], pod)
```

Comment:
Lots of the comments that mentioned `nodeToDaemonPods` references it as "a map from nodes to daemon pods (corresponding to ds) running on the nodes". This is no longer true because target node can be a node that doesn't have the pod running yet. 

Please update all those comments, including `getNodesToDaemonPods`, `manage`, etc. 

We probably need to rename some variables too to avoid confusion, such as `daemonPodsRunning`.

Reply:
Updated comments to `created for nodes`.

Also check DaemonSet's status which is also confused, e.g. `NumberMisscheduled` should be `NumberMiscreated`. We need to update those description when upgrading to beta.